esphome/components/display/display_buffer.h

```diff
@@ -412,6 +433,70 @@ class Image {
   const uint8_t *data_start_;
 };
 
+class HistoryData {
+ public:
+  HistoryData(int length);
+  ~HistoryData();
+  void take_sample(float data);
+  int get_length() const { return length_; }
+  float get_maxvalue() const { return max_; }
+  float get_minvalue() const { return min_; }
+  float get_recent_max() const { return recent_max_; }
+  float get_recent_min() const { return recent_min_; }
+  float get_value(int idx) const { return data_[(count_ + length_ - 1 - idx) % length_]; }
+
+ protected:
+  int length_;
+  int count_;
+  float min_;
+  float max_;
+  float recent_min_;
+  float recent_max_;
+  float *data_;
+};
+
+class Graph : public PollingComponent {
+ public:
+  Graph(int width, int height);
+  ~Graph();
+  virtual bool get_pixel(int x, int y) const;
+  virtual Color get_color_pixel(int x, int y) const;
+  virtual Color get_grayscale_pixel(int x, int y) const;
+  int get_width() const;
+  int get_height() const;
+  void set_sensor(sensor::Sensor *sensor);
+  void set_min_value(float val) { this->min_value_ = val; }
+  void set_max_value(float val) { this->max_value_ = val; }
+  void set_min_range(float val) { this->min_range_ = val; }
+  void set_max_range(float val) { this->max_range_ = val; }
+  void set_line_thickness(uint8_t val) { this->line_thickness_ = val; }
+  void set_line_type(uint8_t val) { this->line_type_ = val; }
+  void set_grid_x(float val) { this->gridspacing_x_ = val; }
+  void set_grid_y(float val) { this->gridspacing_y_ = val; }
+  void set_border(bool val) { this->border_ = val; }
+
+  void update() override;
+  void dump_config() override;
+
+ protected:
+  void redraw_();
+  void set_pixel_(int x, int y);
+  int width_;
+  int height_;
+  float min_value_;
```

Comment:
(3/3) And set these to be NAN by default.
```suggestion
  float min_value_{NAN};
```